### PR TITLE
Index Manager supports restart indexers

### PR DESCRIPTION
**Pull request by thala321**
What are done in this PR
- Fix some user-example so we don't have duplicate tables 
- Refactor index-manager so it's easier to add more code
- Support index manager to restart all indexes when this option INDEX_MANAGER_RESTART_INDEX is set to true
- Add a new field "hash" in the indexer table so it's can be the identifier for the index's config
- Improvement to logger so we don't need to use the env_logger.init()

### CODE REVIEW

**Comment by thala321** [comment]
Screenshot from 2021-08-13 10-06-09

All tests are passed in local with this option enabled
```
INDEX_MANAGER_RESTART_INDEX=true
```

File: index-manager/lib/src/types.rs [hunk 1]
```diff
@@ -35,7 +37,24 @@ pub struct DetailParams {
...
 pub struct IndexConfig {
-    pub config: String,
+    pub config: PathBuf,
     pub mapping: PathBuf,
-    pub schema: String,
+    pub schema: PathBuf,
+    pub identifier: IndexIdentifier,
+}
+
+pub struct IndexIdentifier {
```
Comment (anhhuy0501) [inline index-manager/lib/src/types.rs:46]: Remove redundancy variable

**Review by anhhuy0501 — APPROVED** [review]
LGTM

File: index-manager/lib/src/types.rs (hunk 1 above)
Comment (thala321) [reply index-manager/lib/src/types.rs:46]: I will fix in the next MR. Thanks for the comment

**Pull request merged**